woocommerce/woocommerce-android
# Issue/7166 update order list automatically

Closes: #7166


**We can merge this PR after review, but we need to make sure to merge https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2959 only when the M2 of Tablet support is ready to be merged into trunk to ensure FluxC's trunk and WooAndroid's trunk remain compatible.**

### Description
This PR removes manual refreshes/reloads of the OrderList list and relies on the ListStore component to update the data source automatically.

It also updates FluxC version which contains significant changes - the ListStore component is now informed about any changes to the OrderModel in the database so it can update the data source accordingly. https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2959

This PR contains one regression - not sure if it's a regression to be honest, however, when filters are changed, the order detail in two pane layout isn't automatically updated -> the user needs to select the item they want to open. Considering opening an order detail sends several remote requests I'm not sure this is a true regression - felt quite natural to me.

### Testing instructions

Features to test
- Create order and ensure the new item is available in the list and auto-selected
- Update order status or total price and ensure the list item is automatically updated
- Change filters and ensure they work as expected
- Use search and ensure it works as expcted


### Images/gif

- The only UI facing change is that the list updates are now smoother and faster.
